subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -610,6 +618,56 @@ static int tls_opt_sec_tag_list_get(struct net_context *context,
 	return 0;
 }
 
+static int tls_opt_hostname_set(struct net_context *context,
+				const void *optval, socklen_t optlen)
+{
+	ARG_UNUSED(optlen);
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval) {
+		return -EFAULT;
+	}
+
+	if (mbedtls_ssl_set_hostname(&context->tls->ssl, optval) != 0) {
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int tls_opt_hostname_get(struct net_context *context,
+				void *optval, socklen_t *optlen)
+{
+	char *buf = optval;
+
+	if (!context || !context->tls) {
+		return -EBADF;
+	}
+
+	if (!optval || !optlen) {
+		return -EFAULT;
+	}
+
+	if (*optlen == 0) {
+		return -EINVAL;
+	}
+
+	/* No API to read hostname in mbedTLS, read it directly. */
+	if (context->tls->ssl.hostname == NULL) {
+		buf[0] = '\0';
+		*optlen = 1;
+	} else {
+		strncpy(buf, context->tls->ssl.hostname, *optlen);
+		buf[*optlen - 1] = '\0';
+		*optlen = strlen(buf) + 1;
```

Comment:
If *optlen is 1, then we return also 1 but the string will be empty, which is quite confusing.

Reply:
I think it makes sense, we return empty string which binary size is 1 byte. I'm removing this code anyway, so no use arguing here.